client/client.go

```diff
@@ -759,7 +759,16 @@ func (c *Client) clearPendingOp(op *spb.AFTResult) (*OpResult, error) {
 	if v == nil {
 		return nil, fmt.Errorf("could not dequeue operation %d, unknown operation", op.Id)
 	}
-	delete(c.qs.pendq.Ops, op.Id)
+	
+	actType := c.state.SessParams.GetAckType()
+	operationStatus := op.GetStatus()
+
+	if actType == spb.SessionParameters_RIB_AND_FIB_ACK &&
```

Comment:
ummmm....maybe you want to write the following way? 

if actType == RIB_AND_FIB_ACK {
    switch opStatus := op.GetStatus()
        case: FIB_ACK, FIB_FAILED, FAILED:
            delete(q, op.id)
} else {
    delete(q, op.id)
}